src/Compilers/CSharp/Test/Symbol/Symbols/Metadata/PE/LoadingEvents.cs

```diff
@@ -588,8 +589,61 @@ public void AssociatedField_DuplicateEvents()
 
             Assert.Contains(field.AssociatedSymbol, events);
         }
-        [Flags]
 
+        [Fact, WorkItem(36259, "https://github.com/dotnet/roslyn/issues/36259")]
+        public void BackingFieldSymbolIsPresentInDeclaringTypeForFieldLikeEvent()
+        {
+            var source = @"
+class C
+{
+    public event System.EventHandler E;
+}";
+            var peAssembly = GetAssemblySymbolForReference(CompileToPEReference(source));
+            var declaringType = peAssembly.GlobalNamespace.GetMember<ITypeSymbol>("C");
+
+            var fieldSymbol = Assert.Single(declaringType.GetMembers().OfType<IFieldSymbol>());
+
+            Assert.Same(declaringType.GetMembers().OfType<IEventSymbol>().Single(), fieldSymbol.AssociatedSymbol);
+        }
+
+        [Fact]
+        public void BackingFieldSymbolIsNotPresentInDeclaringTypeForCustomEvent()
+        {
+            var source = @"
+class C
+{
+    public event System.EventHandler E { add { } remove { } }
+}";
+            var peAssembly = GetAssemblySymbolForReference(CompileToPEReference(source));
+            var declaringType = peAssembly.GlobalNamespace.GetMember<ITypeSymbol>("C");
+
+            Assert.Empty(declaringType.GetMembers().OfType<IFieldSymbol>());
+        }
+
+        private static MetadataReference CompileToPEReference(string source)
+        {
+            using var peStream = new MemoryStream();
+
+            Assert.True(CreateCompilation(source).Emit(peStream).Success);
+
+            peStream.Position = 0;
+
+            return MetadataReference.CreateFromStream(peStream);
+        }
+
+        private static IAssemblySymbol GetAssemblySymbolForReference(MetadataReference reference)
+        {
+            var compilation = (Compilation)CSharpCompilation.Create(
+                assemblyName: null,
+                references: new[] { reference },
+                options: new CSharpCompilationOptions(
+                    OutputKind.DynamicallyLinkedLibrary,
+                    metadataImportOptions: MetadataImportOptions.All));
+
+            return (IAssemblySymbol)compilation.GetAssemblyOrModuleSymbol(reference);
+        }
```

Comment:
- Are there existing helpers I should use/places to add tests if I want to specifically test end to end, source to PublicModel.ISymbol

- If not, where should this code go?